datadog/resource_datadog_dashboard.go

```diff
@@ -4641,11 +4704,140 @@ func buildTerraformQueryTableRequests(datadogQueryTableRequests *[]datadogV1.Tab
 			}
 			terraformRequest["cell_display_mode"] = terraformCellDisplayMode
 		}
+		if v, ok := datadogRequest.GetTextFormatsOk(); ok {
+			terraformTextFormats := make([]map[string][]map[string]interface{}, len(*v))
+			for i, textFormat := range *v {
+				test := buildTerraformQueryTableTextFormat(&textFormat)
+				terraformTextFormats[i] = make(map[string][]map[string]interface{})
+				terraformTextFormats[i]["text_format"] = *test
+			}
+			terraformRequest["text_formats"] = terraformTextFormats
+		}
 		terraformRequests[i] = terraformRequest
 	}
 	return &terraformRequests
 }
 
+// Query Table Widget Text Format Helpers
+func buildDatadogQueryTableTextFormat(terraformQueryTableTextFormat *[]interface{}) *[]datadogV1.TableWidgetTextFormatRule {
+	datadogQueryTableTextFormat := make([]datadogV1.TableWidgetTextFormatRule, len(*terraformQueryTableTextFormat))
+	for j, textFormatRule := range *terraformQueryTableTextFormat {
+		terraformTextFormatRule := textFormatRule.(map[string]interface{})
+
+		match, _ := terraformTextFormatRule["match"].([]interface{})
+		terraformTextFormatMatch := match[0].(map[string]interface{})
+		datadogMatch := datadogV1.NewTableWidgetTextFormatMatch(datadogV1.TableWidgetTextFormatMatchType(terraformTextFormatMatch["type"].(string)), terraformTextFormatMatch["value"].(string))
+		datadogTextFormatRule := datadogV1.NewTableWidgetTextFormatRule(*datadogMatch)
+		// Optional
+
+		if v, ok := terraformTextFormatRule["replace"].([]interface{}); ok && len(v) > 0 {
+			if replace, ok := v[0].(map[string]interface{}); ok {
+				if w, ok := replace["type"].(string); ok && len(w) != 0 {
+					switch w {
+					case "all":
+						datadogReplace := datadogV1.NewTableWidgetTextFormatReplaceAll(datadogV1.TABLEWIDGETTEXTFORMATREPLACEALLTYPE_ALL, replace["with"].(string))
+						datadogTextFormatRule.SetReplace(datadogV1.TableWidgetTextFormatReplaceAllAsTableWidgetTextFormatReplace(datadogReplace))
+					case "substring":
+						datadogReplace := datadogV1.NewTableWidgetTextFormatReplaceSubstring(replace["substring"].(string), datadogV1.TABLEWIDGETTEXTFORMATREPLACESUBSTRINGTYPE_SUBSTRING, replace["with"].(string))
+						datadogTextFormatRule.SetReplace(datadogV1.TableWidgetTextFormatReplaceSubstringAsTableWidgetTextFormatReplace(datadogReplace))
+					}
+				}
+			} else {
+				datadogTextFormatRule.Replace = nil
```

Comment:
Would it be ok to remove these nil statements?

Reply:
Second that, could all these = nil statements be removed?